src/lock_file/pypi.rs

```diff
@@ -83,16 +84,30 @@ pub async fn resolve_dependencies<'db>(
         None => (sdist_resolution, PythonLocation::System),
     };
 
+    let mapping = conda_pypi_name_mapping().await?;
+    let packages_to_resolve = conda_python_packages
+        .into_iter()
+        .filter(|pkg| {
+            // skip using the locked package in the case where conda and pypi package has the same name
+            requirements.iter().all(|req| {
+                let has_different_name = req.name != pkg.name.as_str();
+                let found_in_mapping = mapping.get(pkg.name.as_str()).is_some();
```

Comment:
This doesn't work for packages that are not in the map as a one on one. 

E.g.
```
pixi add python-annoy
pixi add --pypi annoy
```
Because this is what it looks like in the map:
```
- conda_name: python-annoy
  import_name: annoy
  mapping_source: regro-bot
  pypi_name: annoy
 ```
You could add this to the same test to see how you can make that work.

Reply:
I added a test in 85a0ca7e61732bceb03cbaca0310cbe5e726b89e

Do we expect `annoy` to appear as both conda and pypi dependency in that case?